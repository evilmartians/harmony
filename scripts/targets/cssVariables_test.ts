import { assertEquals } from "@std/assert";
import { buildCssVars, generateCssVarsForColor } from "./cssVariables.ts";
import { getTestPalette, testExportTarget } from "./testUtils.ts";

Deno.test("generateCssVarsForColor", () => {
  const blue = {
    100: {
      oklch: "oklch(0.1111 0.151562 20)",
      rgbFallback: "#fc8083",
    },
    200: {
      oklch: "oklch(0.2222 0.151562 20)",
      rgbFallback: "#fc8083",
    },
  };

  assertEquals(generateCssVarsForColor("blue", blue, "oklch"), [
    "--blue-100:oklch(0.1111 0.151562 20);",
    "--blue-200:oklch(0.2222 0.151562 20);",
  ]);
  assertEquals(generateCssVarsForColor("blue", blue, "rgbFallback"), [
    "--blue-100:#fc8083;",
    "--blue-200:#fc8083;",
  ]);
});

Deno.test("generateCssVars export target", async (t) => {
  await testExportTarget(
    t,
    buildCssVars,
    Object.keys(getTestPalette())
      .map((color) => `${color}.css`)
      .concat("index.css"),
  );
});
