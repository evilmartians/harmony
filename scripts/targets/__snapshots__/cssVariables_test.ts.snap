export const snapshot = {};

snapshot[`generateCssVars export target 1`] = `
":root {
  --red-100: #fff0ef;
  --red-500: #fc8083;
}

@supports (color: oklch(0% 0 0)) {
  @media (color-gamut: p3) {
    :root {
      --red-100: oklch(0.966797 0.0171875 20);
      --red-500: oklch(0.742188 0.151562 20);
    }
  }
}
"
`;

snapshot[`generateCssVars export target 2`] = `
":root {
  --orange-100: #fff1eb;
}

@supports (color: oklch(0% 0 0)) {
  @media (color-gamut: p3) {
    :root {
      --orange-100: oklch(0.966797 0.0171875 43.3333);
    }
  }
}
"
`;

snapshot[`generateCssVars export target 3`] = `
'@import "./red.css";
@import "./orange.css";'
`;
